docs/development/functional_architecture.md

```diff
@@ -0,0 +1,56 @@
+# Parsec client functional components
+
+Note: This is a work in progress.
+
+- [Parsec client functional components](#parsec-client-functional-components)
+  - [Mountpoint](#mountpoint)
+  - [Gui](#gui)
+  - [Core](#core)
+  - [Metadata server](#metadata-server)
+
+## Mountpoint
+
+The `mountpoint` component provide a virtual filesystem that allow other applications to access its contents.
+This is done using the os filesystem's API that follow native request to the component.
+
+## Gui
+
+The `GUI` component provide to the user a graphical interface and interact directly with the `core` or the os filesystem (but never with the `mountpoint`).
+On contrary to the `mountpoint` component, the `gui` subscribe to events provided by the `metadata server` to notify the user in case of concurrent operation, deletion of a group, etc.
+
+## Core
+
+On top of the 2 previous componenet, the `core` component is responsible for the client side logic and his composed of 5 sub-modules (see figure below):
+
+![Core submodules](../figures/parsec_core_submodules.png)
+
+- The `Virtual File System` (VFS) (`Système de Fichiers Virtuels`): Receive request related to the filesystem originating from the `mountpoint`, thus expose a filesystem like api.
+  To optimize its performance, that component don't push the modifications to the `metadata server`, it only store the encrypted modifications locally on the machine.
+
+- The `Synchronizer` (`Synchroniseur`): That component periodically transfer the locally stored modifications to the `metadata server`.
+  He listen the server for notification about remote modifications and handle version conflict between the local & remote metadata.
+
+- The `Identity manager` (`Gestionnaire d’Identités`): Save the user identity in memory (as a session).
+  The pass-phrase of the user (which isn't stored) concatened with a SALT encrypt the private key of the terminal (`DEVICE_SIG_S_KEY`) and the user private key (`USER_ENC_S_KEY`) (that one is shared between the other terminals used by the user).
+  The `DEVICE_SIG_S_KEY` is used to sign a modification. The `USER_ENC_S_KEY` is used to encrypt user's personal metadata.
```

Comment:
Would it be simpler to only say

```suggestion
  The `DEVICE_SIG_S_KEY` is used to sign a modification. The `USER_ENC_S_KEY` is used to encrypt the user's manifest
```

?

Reply:
:+1: